### PR TITLE
Coca model Inference with PYNATIVE_MODE

### PR DESCRIPTION
**Pull request by Songyuanwei**
# What does this PR do?

Fixes # (issue)

Adds # (feature)
Add Coca mode Inference with PYNATIVE_MODE on ms2.2.10

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/coca/README.md
Comment (zhtmike) [inline examples/coca/README.md:1]: add some inference examples in README

File: examples/coca/configs/coca_vit-l-14.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (zhtmike) [inline examples/coca/configs/coca_vit-l-14.yaml:1]: yaml does not need bracket, following the style in sdv2

File: examples/coca/modules/encoders/image_encoder.py [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+        x = self.layer2(x)
+        x = self.layer3(x)
+        x = self.layer4(x)
+        x = self.attnpool(x)
+
+        return x
+
+    # class MultiheadAttention(nn.Cell):
+    #     def __init__(self, d_model: int, n_head: int, dtype: mstype):
+    #         super(MultiheadAttention, self).__init__()
```
Comment (zhtmike) [inline examples/coca/modules/encoders/image_encoder.py:211]: remove large commented code

**Review by zhtmike — COMMENTED** [review]
think Graph mode is still necessary before merge

File: examples/coca/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+<p align="center">
+  <img width="550" alt="figure" src="https://github.com/mlfoundations/open_clip/assets/52945530/ca75eccb-ea1f-4e9a-b628-1b88b691b849">
+</p>
+<p align="center">
+  <em> Figure 1. Detailed illustration of CoCa architecture and training objectives. [<a href="#references">1</a>] </em>
+</p>
+
+## Get Started
+In this tutorial, we will introduce how to run inference and finetuning experiments using MindONE.
```
Comment (wtomin) [inline examples/coca/README.md:17]: The readme only contains inference examples, not "inference and finetuning experiments".

File: examples/coca/tools/model_conversion/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Model Conversion from Torch to mindspore
+
+We provide scripts of the checkpoint conversion from Torch to MindSpore.
+
+step2. Download the [Official](https://huggingface.co/laion/mscoco_finetuned_CoCa-ViT-L-14-laion2B-s13B-b90k/tree/main), with Mindspore checkpoints.
```
Comment (wtomin) [inline examples/coca/tools/model_conversion/README.md:5]: Step2 -> Step1. Download MindSpore checkpoints -> Download Torch checkpoints.

File: examples/coca/inference.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+        caption = self.tokenizer.decode(generated[0].asnumpy()).split("<|endoftext|>")[0].replace("<|startoftext|>", "")
+
+        return caption
+
+
+if __name__ == "__main__":
+    ms.set_context(mode=ms.PYNATIVE_MODE)
+
+    parser = argparse.ArgumentParser()
```
Comment (wtomin) [inline examples/coca/inference.py:149]: Graph mode is needed. What is the error of running in graph mode?

File: examples/coca/modules/encoders/bpe_simple_vocab_16e6.txt.gz
Comment (SamitHuang) [inline examples/coca/modules/encoders/bpe_simple_vocab_16e6.txt.gz:1]: this vocabulary file is heavily duplicated, e.g. `stable_diffusion_v2/ldm/models/clip/bpe_simple_vocab_16e6.txt.gz`.  Better to re-use it to keep cleanness

File: examples/coca/modules/encoders/text_encoder.py
Comment (SamitHuang) [inline examples/coca/modules/encoders/text_encoder.py:1]: reuse the clip image encoder and text encoder as common modules?

**Pull request closed without merging**